src/main/java/frc/robot/subsystems/shooter/ErectorIOSparkMax.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.subsystems.shooter;
+
+import com.revrobotics.CANSparkBase;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.constants.HardwareIds;
+import frc.robot.util.ThroughBoreEncoder;
+
+public class ErectorIOSparkMax implements ErectorIO {
+  private final CANSparkMax m_leader;
+  private final CANSparkMax m_follower;
+  private final ThroughBoreEncoder m_encoder;
+
+  public ErectorIOSparkMax() {
+    m_leader =
+        new CANSparkMax(HardwareIds.COMP_2024.kErectorLeaderId, CANSparkMax.MotorType.kBrushless);
+    m_follower =
+        new CANSparkMax(HardwareIds.COMP_2024.kErectorFollowerId, CANSparkMax.MotorType.kBrushless);
+    m_encoder =
+        new ThroughBoreEncoder(
+            HardwareIds.COMP_2024.kErectorAbsoluteEncoderPort,
+            HardwareIds.COMP_2024.kErectorRelativeEncoderAPort,
+            HardwareIds.COMP_2024.kErectorRelativeEncoderBPort,
+            Rotation2d.fromRadians(1.1844636870383878 - (Math.PI / 2.0)));
```

Comment:
Configuring the magnetic offset this way is proving to be weird, this needs to be reviewed more